5-appinfra/modules/cicd-pipeline/artifact-registry.tf

```diff
@@ -25,22 +25,18 @@ resource "google_artifact_registry_repository" "container_registry" {
   ]
 }
 
-module "artifact-registry-repository-iam-bindings" {
-  source       = "terraform-google-modules/iam/google//modules/artifact_registry_iam"
-  version      = "~> 7.7"
-  project      = var.project_id
-  repositories = [local.service_name]
-  location     = var.region
-  mode         = "authoritative"
-
-  bindings = {
-    "roles/artifactregistry.reader" = [
-      "serviceAccount:${data.google_project.project.number}-compute@developer.gserviceaccount.com",
-      "serviceAccount:${google_service_account.cloud_deploy.email}",
-      "allAuthenticatedUsers"
-    ],
+resource "google_artifact_registry_repository_iam_member" "member" {
```

Comment:
Is there a requirement to move from the module to a resource?  The module mode can be changed from authoritative to additive.

Reply:
At first I tried changing the module from "authoritative" to "additive", but it started to fail with the following error:

```
│ Error: Invalid for_each argument
│ 
│   on .terraform/modules/app.artifact-registry-repository-iam-bindings/modules/artifact_registry_iam/main.tf line 45, in resource "google_artifact_registry_repository_iam_member" "artifact_registry_iam_additive":
│   45:   for_each   = module.helper.set_additive
│     ├────────────────
│     │ module.helper.set_additive is a set of dynamic, known only after apply
│ 
│ The "for_each" set includes values derived from resource attributes that cannot be determined until apply, and so Terraform cannot determine the
│ full set of keys that will identify the instances of this resource.
│ 
│ When working with unknown values in for_each, it's better to use a map value where the keys are defined statically in your configuration and where
│ only the values contain apply-time results.
│ 
│ Alternatively, you could use the -target planning option to first apply only the resources that the for_each value depends on, and then apply a
│ second time to fully converge.
```

Reply:
Ahh, good to know!  Yeah, given the simple use-case we can probably just use the resource directly for now.